src/main/java/com/artipie/RepoPermissionsFromStorage.java

```diff
@@ -177,11 +176,11 @@ public RepoPermissionsFromStorage(final Storage storage) {
      * @return Completion action with yaml repo section
      */
     private CompletionStage<YamlMapping> repo(final Key key) {
-        return new RxStorageWrapper(this.storage)
-            .value(key)
-            .to(new ContentAsYaml())
-            .map(yaml -> yaml.yamlMapping(RepoPermissionsFromStorage.REPO_SECTION))
-            .to(SingleInterop.get());
+        return SingleInterop.fromFuture(
+            new ConfigFile(key).valueFrom(this.storage)
+        ).to(new ContentAsYaml())
+        .map(yaml -> yaml.yamlMapping(RepoPermissionsFromStorage.REPO_SECTION))
+        .to(SingleInterop.get());
```

Comment:
@genryxy converting `new ConfigFile(...).valueFrom(...)` into `Single` and later converting this `Single` into `CompletionStage` one again seems pointless. I'd suggest to not do it and transform `CompletionStage` into result using `thenApply`, `thenCompose` etc